### PR TITLE
Workflow Executions: Status Service

### PR DESCRIPTION
**Pull request by deepsidhu85**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds in the `status` service which checks the status of a workflow execution every 30 seconds and when in a final state updates the workflow state to this final state. This service is called via a status job which the submission job queues up once a run_id is returned from the submission service

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._
1) Ensure that the tests pass locally

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [X] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

**Review by ericenns — APPROVED** [review]
I tested this against our WES API and it is working as expected. Thanks!

File: test/services/workflow_executions/status_service_test.rb [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+      @workflow_execution = workflow_executions(:irida_next_example_prepared)
+    end
+
+    test 'get status of workflow execution which has completed' do
+      stubs = Faraday::Adapter::Test::Stubs.new
+      stubs.post('/runs') do
+        [
+          200,
+          { 'Content-Type': 'application/json' },
+          { run_id: 'abc123' }
```
Comment (JeffreyThiessen) [inline test/services/workflow_executions/status_service_test.rb:18]: Nitpick: We should use different run id's for the different test files, and maybe for each file too. Something like `status_test_1`

File: app/services/workflow_executions/status_service.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module WorkflowExecutions
+  # Service used to update the status of a WorkflowExecution
+  class StatusService < BaseService
+    CANCELATION_STATES = %w[
```
Comment (JeffreyThiessen) [inline app/services/workflow_executions/status_service.rb:6]: Question: Should these state arrays belong to the API client?

Which strings mean which states is something that could be changed in a version update to the API. Added to a file like `lib/integrations/ga4gh_wes_api/v1/states.rb` might make sense.

**Review by JeffreyThiessen — CHANGES_REQUESTED** [review]
Code looks good. Just 2 comments.

File: app/services/workflow_executions/status_service.rb (hunk 2 above)
Comment (deepsidhu85) [reply app/services/workflow_executions/status_service.rb:6]: Yeah I'm thinking we will move them somewhere else. Just put them inside the service for now and can be updated later once we figure out where we want them. @ericenns thoughts?

File: test/services/workflow_executions/status_service_test.rb (hunk 1 above)
Comment (deepsidhu85) [reply test/services/workflow_executions/status_service_test.rb]: Yeah good idea. Updated in 93a0a6b8e0a98ae38079d25f69a559f3943669d7

File: app/services/workflow_executions/status_service.rb (hunk 2 above)
Comment (deepsidhu85) [reply app/services/workflow_executions/status_service.rb]: I ended up adding the states to the above module in adfc38e07a90093c73bc75f5a0f3707792a7900e

**Review by JeffreyThiessen — APPROVED** [review]
Looks great!

**Pull request merged**